Basic documention for TBA support with some examples (#4102)
redis/jedis: README.md

For the complete failover configuration options and examples, see the [Jedis failover docs](docs/failover.md).

## Token-Based Authentication

> Jedis starts to support Token-Based authentication (TBA) starting with 5.3.0 GA release. This feature is complemented by an extension library that enhances the developer experience as well as providing most of the components required for TBA functionality.

Reviewer: ```suggestion
Jedis supports Token-Based authentication (TBA) starting with 5.3.0 GA release. This feature is complemented by an extension library that enhances the developer experience and provides most of the components required for TBA functionality.
```
